SM7CUSTV1/power.c

```diff
@@ -0,0 +1,224 @@
+#include <libopencm3/stm32/gpio.h>
+#include <libopencmsis/core_cm3.h>
+#include <stdio.h>
+#include <string.h>
+#include <debug.h>
+#include <timer.h>
+#include <power.h>
+#include <common.h>
+#include <board_power.h>
+#include <system.h>
+/* in us */
+#define NODE_CHECK_TIMEOUT	4000
+
+static int node_check(struct power_node const *node)
+{
+	int err = 0;
+	uint32_t port = node->param[0];
+	uint16_t pin = node->param[1];
+
+	timer_start(NODE_CHECK_TIMEOUT);
+
+	while (gpio_get(port, pin) == 0) {
+		if (timer_is_timeout()) {
+			err = -1;
+			break;
+		}
+	}
+	timer_stop();
+
+	return err;
+}
+
+static int node_on(struct power_node *node)
+{
+	int err = 0;
+
+	debug("%s ", node->name);
+
+	if (node->type == NODE_TYPE_ENABLE) {
+		gpio_set(node->param[0], node->param[1]);
+	} else if (node->type == NODE_TYPE_CHECK) {
+		err = node_check(node);
+	} else {
+		power_on_func func = (power_on_func)node->param[0];
+		if (func)
+			err = func();
+	}
+
+	debug("[%c]\n", err ? 'X' : 'O');
+
+	if (err) {
+		node->status = POWER_STATUS_ERR;
+	} else {
+		node->status = POWER_STATUS_ON;
+		if (node->delay)
+			timer_udelay(node->delay);
+	}
+
+	return err;
+}
+
+static void node_off(struct power_node *node)
+{
+	debug("%s\n", node->name);
+
+	if (!strcmp(node->name, "VDD-1.8V")) {
```

Comment:
Why don't pull down these power  domains when power off?

Reply:
This firmware is written with reference to the BM1684xEVB.
in BM1684xEVB
https://github.com/sophgo/mcu/blob/7a7065e58bbcf77f3ba63d53dc69f760cc8f1248/BM1684xEVB/power.c#L67
It is the same.